### ISSUE TITLE
Redundant call API when get dataset's location

### ISSUE THREAD

**Issue by tientt-holistics** [googleapis/google-cloud-ruby#22862]
Getting location from a dataset whose `@gapi` props has `location` should not need to reload the dataset again. 
This is to avoid N+1 issue when getting the locations of all datasets in a project. (Please check the code example below).

#### Environment details

  - OS: Ubuntu 20.04.6 LTS x86_64
  - Ruby version: ruby 2.7.4
  - Gem name and version: google-cloud-bigquery version 1.16.0

#### Steps to reproduce

  1. Get all datasets
  2. Iterate through these datasets and print their locations

#### Code example

```ruby
# this call API GET datasets list https://cloud.google.com/bigquery/docs/reference/rest/v2/datasets/list
# the response already has dataset's location in it
datasets = client.datasets.all 
datasets.each do |dataset|
	puts dataset.location # this will reload the dataset and call API GET dataset https://cloud.google.com/bigquery/docs/reference/rest/v2/datasets/get
end
# after this code block there will be N+1 API calls
```

**Comment by NivedhaSenthil** [comment]
@tientt-holistics Thanks for bringing this into our notice, will be pushing in a fix soon 👍

**Comment by SandeepTuniki** [comment]
The fix is released in version `1.44.2` of google-cloud-bigquery, through #23301.
